pkg/controller/releasemanager/incarnation.go

```diff
@@ -258,43 +258,56 @@ func (i *Incarnation) sync(ctx context.Context) error {
 }
 
 func (i *Incarnation) syncCanaryRules(ctx context.Context) error {
-	return i.controller.applyPlan(ctx, "Sync Canary Rules", &rmplan.SyncAlerts{
-		App:                    i.appName(),
-		Namespace:              i.targetNamespace(),
-		Tag:                    i.tag,
-		Target:                 i.target().Name,
-		ServiceLevelObjectives: i.target().ServiceLevelObjectives,
-		AlertType:              rmplan.Canary,
+	return i.controller.applyPlan(ctx, "Sync Canary Rules", &rmplan.SyncCanaryRules{
+		App:                         i.appName(),
+		Namespace:                   i.targetNamespace(),
+		Tag:                         i.tag,
+		Labels:                      i.defaultLabels(),
+		ServiceLevelObjectiveLabels: i.target().ServiceLevelObjectiveLabels,
+		ServiceLevelObjectives:      i.target().ServiceLevelObjectives,
 	})
 }
 
 func (i *Incarnation) deleteCanaryRules(ctx context.Context) error {
-	return i.controller.applyPlan(ctx, "Delete Canary Rules", &rmplan.DeleteAlerts{
+	return i.controller.applyPlan(ctx, "Delete Canary Rules", &rmplan.DeleteCanaryRules{
 		App:       i.appName(),
 		Namespace: i.targetNamespace(),
 		Tag:       i.tag,
-		AlertType: rmplan.Canary,
 	})
 }
 
-func (i *Incarnation) syncSLIRules(ctx context.Context) error {
-	return i.controller.applyPlan(ctx, "Sync SLI Rules", &rmplan.SyncAlerts{
-		App:                    i.appName(),
-		Namespace:              i.targetNamespace(),
-		Tag:                    i.tag,
-		Target:                 i.target().Name,
-		ServiceLevelObjectives: i.target().ServiceLevelObjectives,
-		AlertType:              rmplan.SLI,
-	})
+func (i *Incarnation) syncTaggedServiceLevels(ctx context.Context) error {
+	if i.picchuConfig.ServiceLevelsFleet != "" && i.picchuConfig.ServiceLevelsNamespace != "" {
```

Comment:
Is it worth logging an error or warning here if they are empty? Might help in future migrations.